jax_healpy/sphtfunc.py

```diff
@@ -328,33 +640,1069 @@ def map2alm(
+        JAX-specific parameter not present in healpy.
+
+    Returns
+    -------
+    map : array-like
+        Random HEALPix map in RING scheme, shape (npix,)
+    alm : array-like, optional
+        Spherical harmonic coefficients in s2fft format (only if alm=True)
+
+    Notes
+    -----
+    This function generates random alm coefficients from the input power spectrum,
+    then transforms them to a map using alm2map with optional smoothing.
+
+    For each (l,m):
+    - m=0: a_l0 is real, drawn from N(0, C_l)
+    - m>0: a_lm is complex, real and imag parts drawn from N(0, C_l/2)
+
+    This function matches the healpy.sphtfunc.synfast API but only supports
+    scalar (non-polarized) output maps. Polarization and pixel window
+    functionality are not yet implemented.
+
+    Examples
+    --------
+    Generate a random map from a power spectrum:
+
+    >>> import jax.numpy as jnp
+    >>> import jax_healpy as jhp
+    >>> lmax = 64
+    >>> ell = jnp.arange(lmax + 1)
+    >>> cl = 1.0 / (ell + 10)**2  # Simple power-law spectrum
+    >>> nside = 32
+    >>> random_map = jhp.synfast(cl, nside, new=True, seed=42, pol=False)
+
+    Generate a map and get the alm coefficients:
+
+    >>> random_map, alm_coeffs = jhp.synfast(cl, nside, alm=True, new=True, seed=42, pol=False)
+    """
+    # Validate unsupported parameters
+    if pol:
+        raise NotImplementedError('pol=True (polarized output) is not supported yet')
+    if pixwin:
+        raise NotImplementedError('pixwin=True is not supported yet')
+    if verbose not in (None, True):
+        warnings.warn('verbose parameter is ignored in JAX implementation', UserWarning)
+    if new is True:
+        warnings.warn('new parameter is ignored in JAX implementation', UserWarning)
+
+    cls = jnp.asarray(cls)
+    cls_lmax = cls.shape[0] - 1
+
+    if lmax is None or lmax < 0:
+        healpy_default = min(cls_lmax, 3 * nside - 1)
+        resolved_lmax = healpy_default
+    else:
+        resolved_lmax = int(lmax)
+
+    min_supported_lmax = 2 * nside - 1
+    if resolved_lmax < min_supported_lmax:
+        raise ValueError(
+            f'synfast requires lmax >= {min_supported_lmax} for nside={nside} when using the '
+            f's2fft backend (got lmax={resolved_lmax}). Provide a longer C_l array or specify a '
+            'larger lmax (you can zero-pad the spectrum) before calling synfast.'
+        )
+
+    alms = _generate_random_alm(cls, resolved_lmax, mmax, prng_key=prng_key)
+
+    # Transform to map with optional smoothing
+    map_synth = alm2map(
+        alms,
+        nside,
+        lmax=resolved_lmax,
+        mmax=mmax,
+        pixwin=pixwin,
+        fwhm=fwhm,
+        sigma=sigma,
+        healpy_ordering=False,
+        method=method,
+    )
+
+    if alm:
+        return map_synth, alms
+    return map_synth
+
+
+@partial(jax.jit, static_argnames=['spin', 'lmax', 'mmax', 'method', 'healpy_ordering'])
```

Comment:
So, this was proposed with iter=0 because the healpy map2alm_spin never got extend to account for non-zero iterations, but I think we should support it
```suggestion
@partial(jax.jit, static_argnames=['spin', 'lmax', 'mmax', 'iter', 'method', 'healpy_ordering'])
```